stdlib/public/SDK/Foundation/JSONEncoder.swift

```diff
@@ -10,6 +10,19 @@
 //
 //===----------------------------------------------------------------------===//
 
+
+fileprivate protocol _JSONStringDictionaryEncodableMarker { }
```

Comment:
Some doc comments about the presence of these protocols would be helpful. 🙂 
Something along the lines of `"A marker protocol used to determine whether a value is a String-keyed dictionary (in which case it should be exempt from key conversion strategies)."` would be great. (Similarly for the decodable marker as well.)